ext/standard/var.c

```diff
@@ -782,13 +782,20 @@ static int php_var_serialize_try_add_sleep_prop(
 	if (Z_TYPE_P(val) == IS_INDIRECT) {
 		val = Z_INDIRECT_P(val);
 		if (Z_TYPE_P(val) == IS_UNDEF) {
-			zend_property_info *info = zend_get_typed_property_info_for_slot(Z_OBJ_P(struc), val);
-			if (info) {
-				zend_throw_error(NULL,
-					"Typed property %s::$%s must not be accessed before initialization (in __sleep)",
-					ZSTR_VAL(Z_OBJCE_P(struc)->name), ZSTR_VAL(error_name));
+			zend_property_info *prop_info = zend_get_typed_property_info_for_slot(Z_OBJ_P(struc), val);
+			if (!prop_info) {
+				return FAILURE;
 			}
-			return FAILURE;
+			zval *prop;
+			if ((prop_info->flags & ZEND_ACC_STATIC) != 0) {
+				prop = &prop_info->ce->default_static_members_table[prop_info->offset];
+			} else {
+				prop = &prop_info->ce->default_properties_table[OBJ_PROP_TO_NUM(prop_info->offset)];
+			}
+			if (Z_ISUNDEF_P(prop) || !ZEND_TYPE_ALLOW_NULL(prop_info->type)) {
+				return SUCCESS;
```

Comment:
If this is going to return success, it might make some sense to add a placeholder such as IS_UNDEF `zend_hash_add(ht, name, undef_zval)` when building this, then check for IS_UNDEF when using the resulting table

Otherwise, __sleep will inconsistently fail to warn about `return ['prop', 'prop'];` when prop is an uninitialized typed property

Reply:
How would that not break on `unserialize()`, since that's the point of this PR?

Reply:
I don't think trying to preserve that warning is worth the complications (using IS_UNDEF for this is not possible -- IS_UNDEF in hashtable means the element doesn't exist).